python/pyspark/ml/wrapper.py

```diff
@@ -118,11 +118,13 @@ def _transfer_params_to_java(self):
         """
         Transforms the embedded params to the companion Java object.
         """
-        paramMap = self.extractParamMap()
         for param in self.params:
-            if param in paramMap:
-                pair = self._make_java_param_pair(param, paramMap[param])
+            if param in self._paramMap:
+                pair = self._make_java_param_pair(param, self._paramMap[param])
                 self._java_obj.set(pair)
+            if param in self._defaultParamMap:
```

Comment:
Should this be an else if? No need to transfer the default value if we've explicitly set it to another value.

Reply:
Sounds reasonable.